src/domain/registration/query.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  QueryClient,
+  useQuery,
+  UseQueryOptions,
+  UseQueryResult,
+} from 'react-query';
+
+import { Registration, RegistrationQueryVariables } from './types';
+import { fetchRegistration } from './utils';
+
+/* istanbul ignore next */
+export const fetchRegistrationQuery = (
+  queryClient: QueryClient,
+  args: RegistrationQueryVariables
+): Promise<Registration> => {
+  return queryClient.fetchQuery('registration', () => fetchRegistration(args));
+};
+
+/* istanbul ignore next */
+export const prefetchRegistrationQuery = (
+  queryClient: QueryClient,
+  args: RegistrationQueryVariables
+): Promise<void> => {
+  return queryClient.prefetchQuery('registration', () =>
```

Comment:
Same here, should id be also passed so registration is uniquely identified from cache?

Reply:
Added registration id to cache key